lib/ProductOpener/Display.pm

```diff
@@ -1116,55 +1118,92 @@ sub display_list_of_tags($$) {
 			];
 	}	
 	
-	eval {
-		$log->debug("Executing MongoDB aggregate query", { query => $aggregate_parameters }) if $log->is_debug();
-		$results = $products_collection->aggregate( $aggregate_parameters );
-	};
-	if ($@) {
-		$log->warn("MongoDB error - retrying once", { error => $@ }) if $log->is_warn();
-		# maybe $connection auto-reconnects but $database and $products_collection still reference the old connection?
-		
-		# opening new connection
+	my $mongodb_query_ref = $aggregate_parameters;
+	
+	my $key = $server_domain . "/" . freeze($mongodb_query_ref);
+	
+	$log->debug("MongoDB aggregate query key", { "key" => $key}) if $log->is_debug();
+
+	use Digest::MD5 qw(md5_hex);
+	$key = md5_hex($key);
```

Comment:
I suppose this is to get a more compact key for Memcached?

Reply:
That's right. The maximum key length for memcached is 250 chars.